### ISSUE TITLE
feature request: line wrapping for tree output

### ISSUE THREAD

**Issue by kwatsen** [mbj4668/pyang#168]
YANG modules with leapref statements can easily exceed the 72 character limit.  For instance:

```
module: ietf-ssh-server-fake
   +--rw fake-ssh-server
      +--rw host-keys
      |  +--rw host-key* [name]
      |     +--rw name           string
      |     +--rw (type)?
      |        +--:(public-key)
      |        |  +--rw public-key?    -> /kc:keychain/private-keys/private-key/name
      |        +--:(certificate)
      |           +--rw certificate?   -> /kc:keychain/private-keys/private-key/certificates/certificate/name {ssh-x509-certs}?
      +--rw client-cert-auth {ssh-x509-certs}?
         +--rw trusted-ca-certs?       -> /kc:keychain/trusted-certificates/name
         +--rw trusted-client-certs?   -> /kc:keychain/trusted-certificates/name
```

The request is to add some ability to line wrap tree output lines.   In the meanwhile, I'm going to write a script to do it...

**Comment by mbj4668** [comment]
If you're writing a script, maybe you can simply update tree.py with your ideas!  ;-)

**Comment by kwatsen** [comment]
silly me, pressed the wrong button a closed it

**Comment by kwatsen** [comment]
Could we add a new pyang tree-output option, something like --generate-key-for-leafrefs that results in output like:

module: ietf-ssh-server-fake
   +--rw fake-ssh-server
      +--rw host-keys
      |  +--rw host-key* [name]
      |     +--rw name           string
      |     +--rw (type)?
      |        +--:(public-key)
      |        |  +--rw public-key?    -> [1]
      |        +--:(certificate)
      |           +--rw certificate?   -> [2]
      +--rw client-cert-auth {ssh-x509-certs}?
         +--rw trusted-ca-certs?       -> [3]
         +--rw trusted-client-certs?   -> [3]

[1] /kc:keychain/private-keys/private-key/name
[2] /kc:keychain/private-keys/private-key/certificates/certificate/name {ssh-x509-certs}?
[3] /kc:keychain/trusted-certificates/name

**Comment by mbj4668** [comment]
Re the original request in this thread, I added an option --tree-max-length options in 4961cad7e.  It is part of the released version 1.7.1.
